services/src/main/scala/cromwell/services/metadata/impl/WriteMetadataActor.scala

```diff
@@ -55,6 +63,43 @@ class WriteMetadataActor(override val batchSize: Int,
     dbAction.map(_ => allPutEvents.size)
   }
 
+  private def sanitizeInputs(
+    metadataWriteActions: NonEmptyVector[MetadataWriteAction],
+    metadataKeysToClean: List[String]
+  ): NonEmptyVector[MetadataWriteAction] =
+    metadataWriteActions.map { metadataWriteAction =>
+      val metadataEvents =
+        metadataWriteAction.events.map { event =>
+          if (
+            metadataKeysToClean.contains(event.key.key)
+            || (event.key.key.contains(":")
+              && metadataKeysToClean.contains(event.key.key.substring(0, event.key.key.indexOf(":"))))
+          ) {
+            event.value match {
+              case Some(_) =>
+                val value = event.value.get
+                MetadataEvent(event.key.copy(),
+                              Option(MetadataValue(StringUtil.cleanUtf8mb4(value.value), value.valueType)),
+                              event.offsetDateTime
+                )
+              case _ => event
+            }
+          } else {
+            event
```

Comment:
I see we have two different levels of conditional that fall through to the unmodified `event` here - can we tidy things up by collapsing this into a single level? That could also save us some cycles by not doing the key string matching in cases where the value is `None`.